src/test/java/solution/EnumBoardTest.java

```diff
@@ -0,0 +1,282 @@
+                                 new EnumBoard(array);
+                             });
+        assertTrue(thrown.getMessage().contains("The sudoku provided is not solvable."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_SudokuIsLegalButNotSolvableBecauseOfSameNumbersInColumnThree() {
+        char[][] array = new char[9][9];
+        array[0] = new char[]{'5', '3', '.', '.', '7', '.', '.', '.', '.'};
+        array[1] = new char[]{'6', '.', '.', '1', '9', '5', '.', '.', '.'};
+        array[2] = new char[]{'.', '9', '8', '.', '.', '.', '.', '6', '.'};
+        array[3] = new char[]{'8', '.', '.', '.', '6', '.', '.', '.', '3'};
+        array[4] = new char[]{'4', '.', '.', '1', '.', '3', '.', '.', '1'};
+        array[5] = new char[]{'7', '.', '.', '.', '2', '.', '.', '.', '6'};
+        array[6] = new char[]{'.', '6', '.', '.', '.', '.', '2', '8', '.'};
+        array[7] = new char[]{'.', '.', '.', '4', '1', '9', '.', '.', '5'};
+        array[8] = new char[]{'.', '.', '.', '.', '8', '.', '.', '7', '9'};
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(array);
+                             });
+        assertTrue(thrown.getMessage().contains("The sudoku provided is not solvable."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_SudokuIsLegalButNotSolvableBecauseOfSameNumbersInBottomRightBox() {
+        char[][] array = new char[9][9];
+        array[0] = new char[]{'5', '.', '.', '.', '7', '.', '5', '.', '.'};
+        array[1] = new char[]{'6', '.', '.', '1', '9', '5', '.', '.', '.'};
+        array[2] = new char[]{'.', '9', '8', '.', '.', '.', '.', '6', '.'};
+        array[3] = new char[]{'8', '.', '.', '.', '6', '.', '.', '.', '3'};
+        array[4] = new char[]{'4', '.', '.', '8', '.', '3', '.', '.', '1'};
+        array[5] = new char[]{'7', '.', '.', '.', '2', '.', '.', '.', '6'};
+        array[6] = new char[]{'.', '6', '.', '.', '.', '.', '2', '8', '.'};
+        array[7] = new char[]{'.', '.', '.', '4', '1', '9', '.', '.', '2'};
+        array[8] = new char[]{'.', '.', '.', '.', '8', '.', '.', '7', '9'};
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(array);
+                             });
+        assertTrue(thrown.getMessage().contains("The sudoku provided is not solvable."));
+    }
+
+    @Test
+    void Should_NotEqual_When_ComparingBoardsWithDifferentChars() {
+        unsolvedArray[0][0] = '.';
+        SudokuBoard second = new EnumBoard(unsolvedArray);
+
+        assertNotEquals(unsolvedBoard, second);
+    }
+
+    @Test
+    void Should_PrintBoard_When_PrintBoardMethodCalled() {
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outContent));
```

Comment:
Dangerous operation; the output stream is set to something else. It's a qustinable practise in general, if it really needs to be used, it should be at least set back.

It actually points to a flaw in design - the board is responsible for printing. It could return a pre-formatted String instead of printing it directly.